src/app/functions/server/onerep.ts

```diff
@@ -439,3 +449,47 @@ export async function fetchAllPages<Data>(
 
   return dataList.flat();
 }
+
+// Local instance map to cache results to prevent excessive API requests
+// Would be nice to share this cache with other pod via Redis in the future
+const profileStatsCache = new Map<string, ProfileStats>();
+export async function getProfilesStats(
+  from: Date,
+  to?: Date,
+): Promise<ProfileStats | undefined> {
+  const fromDateString = from.toISOString().substring(0, 10);
```

Comment:
Not sure if we want to extract this into some sort of `isoFromDate()` helper since we're doing a `.substring(0, 10)` in at least 3 places here.
```js
function isoFromDate(date = new Date()) {
  const [dateString] = date.toISOString().split("T", 1);
    return dateString;
}
```

I think we could also use `Array#unshift()` if you want a one-liner.
But might even reduce this to something like this, since the default `date` is `new Date()` so we wouldn't need the `to ? to... : new Date()...` bits:
```js
const fromDateString = isoFromDate(from);
const toDateString = isoFromDate(to);
```

Conversely
```js
const toDateString = (to ?? new Date()).toISOString().split("T", 1).shift();
```